src/colors.rs

```diff
@@ -1,14 +1,17 @@
+use rgb::{RGB16, RGBA8};
 use std::fmt;
 
-#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+use crate::PngError;
+
+#[derive(Debug, PartialEq, Eq, Clone)]
 /// The color type used to represent this image
 pub enum ColorType {
     /// Grayscale, with one color channel
-    Grayscale,
+    Grayscale { transparent: Option<u16> },
```

Comment:
Need to make sure we are maintaining the documentation for these fields that were moved:

```
/// The palette containing colors used in an Indexed image
/// Contains 3 bytes per color (R+G+B), up to 768
pub palette: Option<Vec<RGBA8>>,
/// The pixel value that should be rendered as transparent
pub transparency_pixel: Option<Vec<u8>>,
```

I think it's also better to keep the name `transparency_pixel`, even though it's longer, the name `transparent` strongly implies it should be a bool when it's actually a pixel value.

Reply:
I kept meaning to add that documentation in there but then kept forgetting. Added now :)

I've changed the name to `transparent_shade` for grey and `transparent_color` for rgb, do you think that will work? I thought it might be good to name them differently, since they have different value types.